### PR TITLE
Create api types package for structured responses

### PR DESCRIPTION
**Pull request by crosbymichael**
This creates a new package `api/types` and also implements a new type for the `/containers/create` endpoint which is also consumed via the client.

### CODE REVIEW

File: api/client/commands.go [hunk 1]
```diff
@@ -2159,23 +2159,19 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 		return nil, err
 	}
 
-	var result engine.Env
-	if err := result.Decode(stream); err != nil {
+	var response *types.ContainerCreateResponse
+	if err := json.NewDecoder(stream).Decode(&response); err != nil {
```
Comment (LK4D4) [inline api/client/commands.go]: You use double pointer tactic here, but not checking for nilness. Looks like potential panic.

**Comment by crosbymichael** [comment]
ping @jfrazelle @tiborvass @LK4D4

File: api/types/types.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package types
+
+// ContainerCreateResponse contains the information returned to a client on the
+// creation of a new container.
+type ContainerCreateResponse struct {
+	// ID is the id of the created container.
+	ID string `json:"Id"`
```
Comment (LK4D4) [inline api/types/types.go]: Why we need such tag? I vote for using default tags, it is simpler.

Comment (tiborvass) [inline api/types/types.go]: +1, it's case insensitive by default: http://play.golang.org/p/jhQA8kZpXr

File: api/types/types.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+	// Warnigns are any warnings encountered during the creation of the container.
```
Comment (tiborvass) [inline api/types/types.go]: `Warnings`

File: api/types/types.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package types
+
+// ContainerCreateResponse contains the information returned to a client on the
+// creation of a new container.
+type ContainerCreateResponse struct {
+	// ID is the id of the created container.
```
Comment (tiborvass) [inline api/types/types.go]: `the ID`

File: api/types/types.go (hunk 2 above)
Comment (crosbymichael) [inline api/types/types.go]: Explicit is better than implicit especially when we want to enforce API consistency and make sure that the tags don't change the responses.

Comment (LK4D4) [inline api/types/types.go]: Hmm, you should be sure by test, not by some string, which also easy to misspell. We had such "100% fact" already in `devicemapper` code with `json: "Id"`.
But I'm okay if you want save them.

**Comment by jfrazelle** [comment]
COOL!

**Comment by tiborvass** [comment]
LGTM

**Comment by estesp** [comment]
LGTM

**Comment by thaJeztah** [comment]
If I understand correctly, this allows defining standardized responses for API calls. Would this be usable for outputing more meaningful messages, such as requested in https://github.com/docker/docker/issues/7761, https://github.com/docker/docker/issues/11016, https://github.com/docker/docker/issues/9120 and https://github.com/docker/docker/issues/9315?

(might be on the wrong foot here :smile:)

**Pull request merged**